wait-for-it.sh

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bash
+# Use this script to test if a given TCP host/port are available
+
+WAITFORIT_cmdname=${0##*/}
+
```

Comment:
Can you explain the addition of this file? you won't be needing this.

Reply:
I see your point. I'll update the code to remove this complexity. I'll have to move the migrate command to be run manually. 

I'm running `python manage.py migrate` when the `web` containers starts.

Even if the `web` container starts after the `db` container, there is no guarantee that the database server has started listening for connections. If the migrate command runs before the connection is established, then we'll get an exception. So we `wait-for-it` (the database server) to start.

More explanation from official docs [here.](https://docs.docker.com/compose/startup-order/)
wait-for-it.sh [GitHub source](https://github.com/vishnubob/wait-for-it)

Reply:
You are using depends_upon in web container, so it won't start until DB container has started.

Reply:
True. The `db` container will always start before `web`. The [official docs](https://docs.docker.com/compose/startup-order/) state:

> However, for startup Compose does not wait until a container is “ready” (whatever that means for your particular application) - only until it’s running. ...

I tried running it without `wait-for-it.sh`, but it failed. MySQL server takes a few seconds to start up/initialize (become ready) after the container starts. 

I will remove both `wait-for-it` and `migrate` commands, because `migrate` needs `wait-for-it`.

Reply:
Right, so what you can do is add a command in Makefile(create makefile if not) which will run migrate on the docker container. So instead of having to wait for the startup, you can run it separately altogether.

Reply:
Interesting. I've never written a Makefile before. I'll have a look, thanks for the suggestion.